flake.nix

```diff
@@ -0,0 +1,169 @@
+    pyk.inputs.nixpkgs.follows = "k-framework/nixpkgs";
+
+  };
+  outputs = { self, k-framework, nixpkgs, flake-utils, poetry2nix
+    , blockchain-k-plugin, rv-utils, pyk }:
+    let
+      overlay = final: prev:
+        let
+          k = k-framework.packages.${prev.system}.k;
+          kavm-deps = prev.stdenv.mkDerivation {
+            pname = "kavm-deps";
+            version = self.rev or "dirty";
+            buildInputs = with prev; [
+              autoconf
+              automake
+              cmake
+              clang
+              cryptopp.dev
+              gmp
+              openssl.dev
+              pkg-config
+              procps
+              libtool
+            ];
+
+            src = prev.stdenv.mkDerivation {
+              name = "kavm-deps-${self.rev or "dirty"}-src";
+              phases = [ "installPhase" ];
+              src = ./Makefile;
+              installPhase = ''
+                mkdir $out
+                cp $src $out/Makefile
+                chmod -R u+w $out
+                mkdir -p $out/deps/plugin
+                cp -rv ${prev.blockchain-k-plugin-src}/* $out/deps/plugin/
+              '';
+            };
+
+            dontUseCmakeConfigure = true;
+
+            buildPhase = ''
+              make SHELL=${prev.bash}/bin/bash plugin-deps
+            '';
+
+            enableParallelBuilding = true;
+
+            installPhase = ''
+              mkdir -p $out
+              mv .build/usr/* $out/
+            '';
+          };
+
+          kavm-bin = prev.poetry2nix.mkPoetryApplication {
+            python = prev.python310;
+            projectDir = ./kavm;
+            overrides = prev.poetry2nix.overrides.withDefaults
+              (finalPython: prevPython: { pyk = prev.python310Packages.pyk; });
+            groups = [ ];
+            # We remove `"dev"` from `checkGroups`, so that poetry2nix does not try to resolve dev dependencies.
+            checkGroups = [ ];
+            propagatedBuildInputs = [ k prev.llvm-backend ];
+          };
+        in {
+          kavm = prev.stdenv.mkDerivation {
+            pname = "kavm";
+            version = self.rev or "dirty";
+            buildInputs = with prev; [
+              kavm-bin
+              cryptopp.dev
+              curl.dev
+              secp256k1
+              msgpack
+              openssl.dev
+              procps
+            ];
+
+            src = prev.nix-gitignore.gitignoreSourcePure [
+              ./.gitignore
+              ".github/"
+              "result*"
+              "*.nix"
+              "deps/"
+              "kavm/"
+            ] ./.;
+
+            dontUseCmakeConfigure = true;
+
+            postPatch = ''
+              substituteInPlace ./Makefile \
+                --replace '$(MAKE) build -C $(PY_KAVM_DIR)' 'echo "Skip py-kavm"' \
+                --replace 'py-kavm $(VENV_DIR)/pyvenv.cfg' ' ' \
+                --replace 'plugin-deps $(hook_includes)' '$(hook_includes)'
+            '';
```

Comment:
this is a bot hacky. basically I am removing any dependencies on `py-kavm` from the makefile as we build it separately using `poetry2nix` perhaps a refactor would be in order such that we can disable building of `py-kavm` via the Makefile through a flag. Same thing for `plugin-deps`, i.e. we build beforehand and don't want the Makefile to try to re-build again.

Reply:
Well, I would like to move towards a future with less Makefile and more kbuild + Nix. So I think for now a hacky solution is OK.